vllm/entrypoints/openai/cli_args.py

```diff
@@ -174,13 +174,21 @@ def make_arg_parser(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
     parser.add_argument(
         "--tool-call-parser",
         type=str,
-        choices=["mistral", "hermes"],
```

Comment:
I think we still want to specify the choices of `["mistral", "hermes", "internlm2_5"]`, but make this _optional_ in the event that `--enable-auto-tool-choice` is called with `--tool-parser-plugin`. 

It would be good for people to know which tool call parsers are available by default, and this makes sure that the expected values get into the auto-generated documentation.

Reply:
if special a choices in the `add_argument`,  use cannot special a `--tool-call-parser` which register in the `--tool-parser-plugin`.

maybe we can given the default choices to the help information.

Reply:
I think what I'm trying to say is that you could keep the choices of `["mistral", "hermes", "internlm2_5"]` and then do something like this in `serving_chat.py`:

Current state:
```python
if self.enable_auto_tools:
    try:
        self.tool_parser = ToolParserManager.get_tool_parser(tool_parser)
    except Exception as e:
        raise TypeError("Error: --enable-auto-tool-choice requires tool_parser:'{tool_parser}' which has not  been registered") from e
```

Possible changes:
```python
# if a plugin is not specified; we can do this already
if self.enable_auto_tools and not self.tool_parser_plugin:
    plugin_name = tool_parser # one of the options from the CLI argument, e.g. hermes or mistral

# if a plugin is specified - this may require some refactoring to get the tool parser plugin loaded in serving chat
elif self.enable_auto_tools and self.tool_parser_plugin:
    # get the name of the plugin loaded from `--tool-parser-plugin`
    plugin_name = get_plugin_name_somehow_from_loaded_plugin()

# handle additional cases here
try: 
    self.tool_parser = ToolParserManager.get_tool_parser(plugin_name)
except Exception as e:
    raise TypeError("You must specify a valid value for --tool-call-parser OR a value tool parser plugin"
```

Reply:
but In my design, a plugin can register any number of tool parsers into vllm, and user use `--tool-call-parser` to specify the one he want to use just like the default tool parser write in vllm.

Reply:
Ohhhh, I see. Hmm. I'm not sure what the best pattern would be for the arguments here, then. @DarkLight1337 @mgoin do y'all have any thoughts?

Reply:
Maybe we can use metavar instead of choices to display help information.
```
    valid_tool_parsers = ["mistral", "hermes", "internlm2", "internlm2_5"]
    parser.add_argument(
        "--tool-call-parser",
        type=str,
        metavar=
        "{" + ",".join(valid_tool_parsers) +  "} or name registered in "
        "--tool-parser-plugin",
        default=None,
        help=
        "Select the tool call parser depending on the model that you're using."
        " This is used to parse the model-generated tool call into OpenAI API "
        "format. Required for --enable-auto-tool-choice.")
```

the help will look like this:
```
--tool-call-parser {mistral,hermes,internlm2,internlm2_5} or name registered in --tool-parser-plugin
                        Select the tool call parser depending on the model
                        that you're using. This is used to parse the model-
                        generated tool call into OpenAI API format. Required
                        for --enable-auto-tool-choice.
```

and move the plugin import and tool call parser check to run_server to check the invalid tool call parser name quickly.(before the model loads).

```
if args.tool_parser_plugin and len(args.tool_parser_plugin) > 3:
        ToolParserManager.import_tool_parser(args.tool_parser_plugin)
    
    if args.enable_auto_tool_choice:
        if args.tool_call_parser not in ToolParserManager.tool_parsers.keys():
            raise KeyError(f"invalid tool call parser: {args.tool_call_parser} "
                        f"chose from {{ {','.join(ToolParserManager.tool_parsers.keys())} }}")

```

error info look like this:

```
Traceback (most recent call last):
  File "/LocalRun/jun.dai/conda/envs/vllm_env/bin/vllm", line 8, in <module>
    sys.exit(main())
  File "/LocalRun/jun.dai/code/github/sydnash/vllm/vllm/scripts.py", line 165, in main
    args.dispatch_function(args)
  File "/LocalRun/jun.dai/code/github/sydnash/vllm/vllm/scripts.py", line 37, in serve
    asyncio.run(run_server(args))
  File "/LocalRun/jun.dai/conda/envs/vllm_env/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/LocalRun/jun.dai/conda/envs/vllm_env/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/LocalRun/jun.dai/code/github/sydnash/vllm/vllm/entrypoints/openai/api_server.py", line 505, in run_server
    raise KeyError(f"invalid tool call parser: {args.tool_call_parser} "
KeyError: 'invalid tool call parser: internlm3 chose from { hermes,internlm2,internlm2_5,mistral,internlm }'
```